File: segundo/1er cuatri/PDOO/irrgarten_ruby/Irrgarten/monster.rb
# frozen_string_literal: true
require_relative 'dice'
require_relative 'labyrinth_character'

class Monster < LabyrinthCharacter
  @@INITIAL_HEALTH = 5

  def initialize(name, intelligence, strength)
    super(name, intelligence, strength, @@INITIAL_HEALTH)
  end

  def attack
    Dice.intensity(@strength)
  end

  def defend(received_attack)
    is_dead = dead

    unless is_dead
      defensive_energy = Dice.intensity(@intelligence)

      if defensive_energy < received_attack
        got_wounded
        is_dead = dead
      end
    end

    is_dead
  end

  def to_s
    "[#{super.to_s} ]\n"
  end

  def inspect
    to_s
  end
end

File: segundo/1er cuatri/PDOO/Irrgarten_java/src/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alvarogz
 */
public class Monster extends LabyrinthCharacter{
    private static final int INITIAL_HEALTH = 5;
    
    public Monster(String name, float intelligence, float strength) {
        super(name, intelligence, strength, INITIAL_HEALTH);
    }
    
    @Override
    public float attack() {
        return Dice.intensity(super.getStrength());
    }
    
    @Override
    public boolean defend(float receivedAttack) {
        boolean isDead = dead();
        
        if (!isDead){
            float defensiveEnergy = Dice.intensity(super.getIntelligence());
            
            if (defensiveEnergy < receivedAttack){
                gotWounded();
                isDead = dead();
            }
        }
        
        return isDead;
    }
    
    @Override
    public String toString() {
        return "[" + super.toString() + "]" + "\n";
    }
 }
